src/main/java/ro/chirila/programarispital/controller/AppointmentController.java

```diff
@@ -0,0 +1,50 @@
+
+package ro.chirila.programarispital.controller;
+
+import jakarta.transaction.Transactional;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.chirila.programarispital.repository.dto.AppointmentRequestDTO;
+import ro.chirila.programarispital.repository.dto.AppointmentResponseDTO;
+import ro.chirila.programarispital.repository.dto.AppointmentUpdateDTO;
+import ro.chirila.programarispital.service.AppointmentService;
+import ro.chirila.programarispital.service.SendEmailService;
+
+import java.util.concurrent.CompletableFuture;
+
+@CrossOrigin(origins = "*", maxAge = 3600)
+@RestController
+@RequestMapping("/appointment")
+public class AppointmentController {
+
+    private final AppointmentService appointmentService;
+    private final SendEmailService sendEmailService;
+
+    public AppointmentController(AppointmentService appointmentService, SendEmailService sendEmailService) {
+        this.appointmentService = appointmentService;
+        this.sendEmailService = sendEmailService;
+    }
+    @PostMapping("/create-appointment")
+    @Transactional
```

Comment:
I don't think you need Transactional at endpoint level. Make your services transactional, either at class level or method level (at class level it makes all the methods transactional). See also: https://www.baeldung.com/transaction-configuration-with-jpa-and-spring